src/types/CurrentUser.ts

```diff
@@ -74,106 +83,48 @@ export type CurrentUser = {
-            console.log('Failed at isRejoining');
-            return false;
-        }
-
-        if (!('platform' in obj) || (obj.platform !== undefined && typeof obj.platform !== 'string' && obj.platform !== null)) {
-            console.log('Failed at platform');
-            return false;
-        }
-
-        if (!('profilePicOverride' in obj) || (obj.profilePicOverride !== undefined && typeof obj.profilePicOverride !== 'string' && obj.profilePicOverride !== null)) {
-            console.log('Failed at profilePicOverride');
-            return false;
-        }
-
-        if (!('status' in obj) || (obj.status !== undefined && typeof obj.status !== 'string' && obj.status !== null)) {
-            console.log('Failed at status');
-            return false;
-        }
+    try {
+        if (('displayName' in presenceObj) && (presenceObj.displayName !== undefined && typeof presenceObj.displayName !== 'string')) errorList.push('displayName');
+        if (('avatarThumbnail' in presenceObj) && (presenceObj.avatarThumbnail !== undefined && typeof presenceObj.avatarThumbnail !== 'string')) errorList.push('avatarThumbnail');
+        if (('groups' in presenceObj) && (presenceObj.groups !== undefined && (!Array.isArray(presenceObj.groups) || !presenceObj.groups.every(group => typeof group === 'string')))) errorList.push('groups');
+        if (('id' in presenceObj) && (presenceObj.id !== undefined && typeof presenceObj.id !== 'string')) errorList.push('id');
+        if (('instance' in presenceObj) && (presenceObj.instance !== undefined && typeof presenceObj.instance !== 'string')) errorList.push('instance');
+        if (('instanceType' in presenceObj) && (presenceObj.instanceType !== undefined && typeof presenceObj.instanceType !== 'string')) errorList.push('instanceType');
+        if (('isRejoining' in presenceObj) && (presenceObj.isRejoining !== undefined && typeof presenceObj.isRejoining !== 'string')) errorList.push('isRejoining');
+        if (('platform' in presenceObj) && (presenceObj.platform !== undefined && typeof presenceObj.platform !== 'string')) errorList.push('platform');
+        if (('profilePicOverride' in presenceObj) && (presenceObj.profilePicOverride !== undefined && typeof presenceObj.profilePicOverride !== 'string')) errorList.push('profilePicOverride');
+        if (('status' in presenceObj) && (presenceObj.status !== undefined && typeof presenceObj.status !== 'string')) errorList.push('status');
+        if (('travelingToInstance' in presenceObj) && (presenceObj.travelingToInstance !== undefined && typeof presenceObj.travelingToInstance !== 'string')) errorList.push('travelingToInstance');
+        if (!('travelingToWorld' in presenceObj && typeof presenceObj.travelingToWorld === 'string')) errorList.push('travelingToWorld');
+        if (!('world' in presenceObj && typeof presenceObj.world === 'string')) errorList.push('world');
```

Comment:
Ouch. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof

Reply:
I know you want to do the error list, but in this case it would be worth taking a look at some other way to do that. There's so many ways to do such thing. You could do simply with [zod](https://zod.dev), or this all could be done simply with a Object.entries as well.